### PR TITLE
Add password handling

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: cloudify_awssdk/ec2/resources/instances.py [hunk 1]
```diff
@@ -412,3 +432,37 @@ def _handle_userdata(existing_userdata):
             [existing_userdata, install_agent_userdata])
 
     return final_userdata
+
+
+def _handle_password(iface):
+    if not ctx.node.properties['use_password']:
+        return True
+    key_data = ctx.node.properties['agent_config'].get('key')
```
Comment (mabuaisha) [inline cloudify_awssdk/ec2/resources/instances.py]: Are we sure that `agent_config` will be always provided ?

File: cloudify_awssdk/ec2/decrypt.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if endianness == 'little':
+        # see http://stackoverflow.com/a/931095/309233
+        s = s[::-1]
+
+    return s
+
+
+def decrypt_password(rsaKey, password):
+    # Undo the whatever-they-do to the ciphertext to get the integer
+    encryptedData = base64.b64decode(password)
```
Comment (mabuaisha) [inline cloudify_awssdk/ec2/decrypt.py]: I think we may need to check if both `rsaKey` and `password` are set or not

Moreover I think it is better to change `rsaKey` to `rsa_key` to match the python convention

File: cloudify_awssdk/ec2/resources/instances.py (hunk 1 above)
Comment (EarthmanT) [reply cloudify_awssdk/ec2/resources/instances.py]: nope. that's why we check for a key pair attached via a relationship.

File: cloudify_awssdk/ec2/decrypt.py (hunk 2 above)
Comment (EarthmanT) [reply cloudify_awssdk/ec2/decrypt.py]: rsakey and password are both unnamed keywords. so they have to be provided.

changed.

Comment (mabuaisha) [reply cloudify_awssdk/ec2/decrypt.py]: 👍

File: cloudify_awssdk/ec2/resources/instances.py (hunk 1 above)
Comment (mabuaisha) [reply cloudify_awssdk/ec2/resources/instances.py]: 👍

**Pull request merged**